libraries/AP_Frsky_Telem/AP_Frsky_Telem.cpp

```diff
@@ -695,6 +695,8 @@ uint32_t AP_Frsky_Telem::calc_ap_status(void)
     ap_status |= (uint8_t)(AP_Notify::flags.failsafe_battery)<<AP_BATT_FS_OFFSET;
     // bad ekf flag
     ap_status |= (uint8_t)(AP_Notify::flags.ekf_bad)<<AP_EKF_FS_OFFSET;
+    // IMU temperature: offset -19, 0 means temp =< 19°, 63 means temp => 82°
+    ap_status |= ((AP::ins().get_temperature(0) < AP_IMU_TEMP_ORIGIN) ? 0 : (AP::ins().get_temperature(0) > (AP_IMU_TEMP_ORIGIN + AP_IMU_TEMP_LIMIT)) ? AP_IMU_TEMP_LIMIT : (uint8_t)roundf(AP::ins().get_temperature(0) - AP_IMU_TEMP_ORIGIN))<<AP_IMU_TEMP_OFFSET;
```

Comment:
this line need simplification, it is very hard to follow what it does. Can you please break this up into multiple lines?

Reply:
also, don't call get_temperature() 3 times

Reply:
your second request is done and committed